spec/controllers/api/v1/providers_controller_spec.rb

```diff
@@ -1,66 +1,146 @@
 require 'rails_helper'
 
 describe API::V1::ProvidersController, type: :controller do
-  describe "index" do
-    it "calls limit on the model with default value of 100" do
-      allow(controller).to receive(:authenticate)
-      expect(Provider).to receive_message_chain(:changed_since, :limit).with(100).and_return([])
+  describe '#index' do
+    RSpec::Matchers.define :have_providers do |*providers|
+      def provider_codes(body)
+        json = JSON.parse(body)
+        json.map { |provider| provider["institution_code"] }
+      end
 
-      get :index
+      match do |body|
+        provider_codes(body) == providers.map(&:provider_code)
+      end
+
+      failure_message do |body|
+        <<~STRING
+          expected provider codes #{providers.map(&:provider_code)}
+            to be found in body #{provider_codes(body)}
+        STRING
+      end
     end
 
-    it 'renders a 400 when the changed_since param is not valid' do
+    RSpec::Matchers.define :have_empty_response do
```

Comment:
I think that having `have_empty_response` is a bit generic (what kind of response?), and probably duplicates existing functionality (e.g. `expect(JSON.parse(response.body)).to eq []`). I know that this is just copied over from the `courses_controller_spec.rb`, I played around with that tonight and came up with something that's a bit more generic that can be used as:

```ruby
should have_courses(old_course, new_course) # existing functionality
should have_courses                         # true if any courses are found
should_not have_courses                     # true if no courses are found
```

Here's what it looks like, it would be easy to translate to providers if you think it works:

```ruby
RSpec::Matchers.define :have_courses do |*courses|
      def course_codes(body)
        json = JSON.parse(body)
        json.map { |course| course["course_code"] }
      end

      match do |body|
        if courses.any?
          course_codes(body) == courses.map(&:course_code)
        else
          course_codes(body).any?
        end
      end

      failure_message do |body|
        if courses.any?
          <<~STRING
            expected course codes #{courses.map(&:course_code)}
              to be found in body #{course_codes(body)}
          STRING
        else
          'expected courses to be present, but no courses found'
        end
      end

      failure_message_when_negated do |body|
        if courses.any?
          <<~STRING
              expected course codes #{courses.map(&:course_code)}
            not to be found in body #{course_codes(body)}
          STRING
        else
          'expected no courses to be present, but some are'
        end
      end
    end
```

Reply:
If we do keep `have_empty_response` then it should probably be moved into `spec/support` and included into `spec-helper.rb`

Reply:
done the alteration for negated, left DRY for later, made a note on the trello card.